src/manifest.c

```diff
@@ -323,6 +322,20 @@ static struct manifest *manifest_from_file(int version, char *component, bool he
 		c = c2;
 
 		file->filename = strdup_or_die(c);
+		file->bundlename = strdup_or_die(c);
```

Comment:
Just an idea for you, if you like to play with strings. In this line you are allocating a string as big as filename for bundlename, but this will leave some trailing bytes, becase the bundlename length is smaller than filename length. So, instead of using strdup you could use strndup:

    ptr = strrchr(filename, '.');
    bundlename = strndup(filename, ptr - filename - 2);

And by that you won't need to look the string for the 'I' again.

Reply:
Nice!